ims/einstein/operations.py

```diff
@@ -211,105 +243,130 @@ def shutdown(self):
         self.db.close()
 
     # Provisions from HIL and Boots the given node with given image
+
     @log
-    def provision(self, node_name, img_name, nic):
-        try:
-            mac_addr = "01-" + self.hil.get_node_mac_addr(node_name). \
-                replace(":", "-")
+    def create_disk(self, disk_name, img_name):
+        """
+        Creates a new image named <disk_name> from <img_name>,
+        and then creates an iscsi endpoint for <disk_name>.
+        Returns the disk image name (which also happens to be the iscsi target
+        name)
 
+        Note: The name of the iscsi target and the name of disk in ceph is same
+        """
+        # Database Operations
+        try:
+            # Find the image id by using the image name, and then create a new
+            # entry for the new disk image.
             parent_id = self.db.image.fetch_id_with_name_from_project(img_name,
                                                                       self.proj
                                                                       )
-            self.db.image.insert(node_name, self.pid, parent_id)
-            clone_ceph_name = self.__get_ceph_image_name(node_name)
+            self.db.image.insert(disk_name, self.pid, parent_id)
+
+            # This generates the name that BMI *must* have used when it created
+            # the copy in ceph of img_name
             ceph_img_name = self.__get_ceph_image_name(img_name)
+
+            # Prepare the ceph image name to be used.
+            clone_ceph_name = self.__get_ceph_image_name(disk_name)
+        except db_exceptions.ORMException as e:
+            # Quick solution for returning the name
```

Comment:
We can probably delete these comments. Also it seems like an `ORMException` will be triggered if someone tries to import a different image with the same name & project id.  That seems ok to me, just want to make sure that behaviour is fine.